contracts/staking/RewardsDistributorAdmin.sol

```diff
@@ -0,0 +1,83 @@
+pragma solidity ^0.8.0;
+
+import "./TribalChief.sol";
+import "../refs/CoreRef.sol";
+import "../external/rari/IRewardsDistributor.sol";
+
+contract RewardsDistributorAdmin is IRewardsDistributor, CoreRef {
+    /// rewards distributor contract address
+    IRewardsDistributor public rewardsDistributorContract;
+
+    constructor(
+        address coreAddress,
+        IRewardsDistributor _rewardsDistributorContract
+    ) CoreRef(coreAddress) {
+        rewardsDistributorContract = _rewardsDistributorContract;
+        // Create new admin role for this contract
+        // grant this role to all AutoRewardsDistributors so that they can call 
+        _setContractAdminRole(keccak256("REWARDS_DISTRIBUTOR_ADMIN_ROLE"));
+    }
+
+    /**
+      * @notice Begins transfer of admin rights. The newPendingAdmin must call `_acceptAdmin` to finalize the transfer.
+      * @dev Admin function to begin change of admin. The newPendingAdmin must call `_acceptAdmin` to finalize the transfer.
+      * @param newPendingAdmin New pending admin.
+      */
+    function _setPendingAdmin(address newPendingAdmin) external override onlyGovernorOrAdmin {
```

Comment:
This should be exclusively `onlyGovernor`, admin shouldn't be able to change this contract